src/Shared/Components/EditImageFormField/EditImageFormField.tsx

```diff
@@ -0,0 +1,204 @@
+import { SyntheticEvent, useState } from 'react'
+import { toast } from 'react-toastify'
+import { showError } from '@Common/Helper'
+import { CustomInput } from '@Common/CustomInput'
+import { ButtonWithLoader, ImageWithFallback } from '@Shared/Components'
+import { validateIfImageExist, validateURL } from '@Shared/validations'
+import { ReactComponent as ICPencil } from '@Icons/ic-pencil.svg'
+import { EditImageFormFieldProps, FallbackImageProps } from './types'
+import { DEFAULT_IMAGE_DIMENSIONS, DEFAULT_MAX_IMAGE_SIZE, EMPTY_PREVIEW_URL_ERROR_MESSAGE } from './constants'
+import './EditImageFormField.scss'
+
+const FallbackImage = ({ showEditIcon, defaultIcon }: FallbackImageProps) => (
+    <div
+        className={`flex dc__align-self-start dc__no-shrink br-4 edit-image-form-field__fallback-image p-12 ${showEditIcon ? 'base-image' : ''}`}
+        // Adding inline style to make sure it is configurable through constants
+        style={{
+            height: DEFAULT_IMAGE_DIMENSIONS.height,
+            width: DEFAULT_IMAGE_DIMENSIONS.width,
+        }}
+    >
+        {defaultIcon}
+    </div>
+)
+
+// NOTE: Have to replace component in UpsertTenantModal with EditImageFormField when prioritized.
+const EditImageFormField = ({
+    defaultIcon,
+    errorMessage,
+    handleError,
+    url,
+    handleURLChange,
+    ariaLabelPrefix,
+    dataTestIdPrefix,
+    altText,
+}: EditImageFormFieldProps) => {
+    const [lastPreviewedURL, setLastPreviewedURL] = useState<string>(url)
+    const [isEditing, setIsEditing] = useState<boolean>(false)
+    const [isLoading, setIsLoading] = useState<boolean>(false)
+    // Since we need to show error message for empty preview URL but not propagate it to parent, we need to maintain a separate state
+    const [emptyPreviewURLErrorMessage, setEmptyPreviewURLErrorMessage] = useState<string>('')
+
+    const handleEnableEditing = () => {
+        setIsEditing(true)
+    }
+
+    const handleLastPreviewedURLChange = (newURL: string) => {
+        setLastPreviewedURL(newURL)
+    }
+
+    const handleReset = (newURL?: string) => {
+        handleLastPreviewedURLChange(newURL ?? lastPreviewedURL)
+        handleURLChange(newURL ?? lastPreviewedURL)
+        handleError('')
+        setIsEditing(false)
+        setIsLoading(false)
+    }
+
+    const handleSuccess = () => {
+        handleReset(url)
+    }
+
+    const handleCancel = () => {
+        handleReset()
+    }
+
+    const handleChange = (event: SyntheticEvent) => {
+        const { value } = event.target as HTMLInputElement
+        handleURLChange(value)
+        if (!value.trim()) {
+            handleError('')
+            return
+        }
+        setEmptyPreviewURLErrorMessage('')
+        handleError(validateURL(value, false).message)
```

Comment:
validateURL also correctly parses ssh, so maybe should add protocol validation?

Reply:
can add a todo to pick later; not required as of now